addons/udes_stock_cron/models/stock_picking.py

```diff
@@ -0,0 +1,387 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models, _, api
+from collections import defaultdict
+from odoo.exceptions import UserError, ValidationError
+from psycopg2 import OperationalError, errorcodes
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+PG_CONCURRENCY_ERRORS_TO_RETRY = (
+    errorcodes.LOCK_NOT_AVAILABLE,
+    errorcodes.SERIALIZATION_FAILURE,
+    errorcodes.DEADLOCK_DETECTED,
+)
+MAX_TRIES_ON_CONCURRENCY_FAILURE = 5
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    batch_id = fields.Many2one("stock.picking.batch", copy=True)
```

Comment:
Perhaps this should be in udes_stock - i guess this was to get something to work in tests?